### PR TITLE
Support for the google wrappers

### PR DESCRIPTION
**Pull request by olegbespalov**
# What?

This change allows the use of the [Google Protobuff wrappers](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/wrappers.proto) and brings the tests for that functionality.

# Why?

Initially reported in https://github.com/grafana/k6/issues/3232 and attempted to fix in https://github.com/grafana/k6/pull/3238

### CODE REVIEW

File: grpc/testdata/wrappers_testing/service.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+func (s *service) TestInteger(ctx context.Context, in *wrappers.Int64Value) (*wrappers.Int64Value, error) {
+	if s.TestIntegerImplementation != nil {
+		return s.TestIntegerImplementation(ctx, in)
+	}
+
+	return s.UnimplementedServiceServer.TestInteger(ctx, in)
+}
+
+func (s *service) TestBoolean(ctx context.Context, in *wrappers.BoolValue) (*wrappers.BoolValue, error) {
+	if s.TestBooleanImplementation != nil {
```
Comment (mstoykov) [inline grpc/testdata/wrappers_testing/service.go:48]:
```suggestion
	if s.TestDoublemplementation != nil {
```
?

File: grpc/testdata/wrappers_testing/service.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		return s.TestBooleanImplementation(ctx, in)
+	}
+
+	return s.UnimplementedServiceServer.TestBoolean(ctx, in)
+}
+
+func (s *service) TestDouble(ctx context.Context, in *wrappers.DoubleValue) (*wrappers.DoubleValue, error) {
+	if s.TestBooleanImplementation != nil {
```
Comment (mstoykov) [inline grpc/testdata/wrappers_testing/service.go:56]:
```suggestion
	if s.TestDoubleImplementation != nil {
```
?

Comment (olegbespalov) [reply grpc/testdata/wrappers_testing/service.go:56]: :facepalm:

File: lib/netext/grpcext/stream.go [hunk 3]
```diff
@@ -78,15 +78,15 @@ func (s *Stream) receive() (*dynamicpb.Message, error) {
 // {"x":6,"y":4}
 // rather than the desired:
 // {"x":6,"y":4,"z":0}
-func (s *Stream) convert(msg *dynamicpb.Message) (map[string]interface{}, error) {
+func (s *Stream) convert(msg *dynamicpb.Message) (interface{}, error) {
```
Comment (mstoykov) [inline lib/netext/grpcext/stream.go:81]: nit: can we make this into a function taking the marshaller and reusing it in both places?

**Review by mstoykov — APPROVED** [review]

**Comment by mstoykov** [comment]
LGTM otherwise then us having what seems like hte same code in two places

File: lib/netext/grpcext/stream.go [hunk 4]
```diff
@@ -68,8 +68,8 @@ func (s *Stream) receive() (*dynamicpb.Message, error) {
 //
 //	message Point {
 //	   double x = 1;
-//		  double y = 2;
-//		  double z = 3;
+//		double y = 2;
+//		double z = 3;
```
Comment (mstoykov) [inline lib/netext/grpcext/stream.go:72]: nit: not certain what happened here, but I can't propose a change through the UI that seems right :(

**Review by mstoykov — APPROVED** [review]
LGTM :+1:

There is a small comment problem :(

**Review by codebien — COMMENTED** [review]
LGTM, waiting for the rebase + nit fixes for final approval

File: lib/netext/grpcext/stream.go [hunk 5]
```diff
@@ -60,16 +60,16 @@ func (s *Stream) receive() (*dynamicpb.Message, error) {
...
 // there is a lot of marshaling/unmarshaling here, but if we just pass the dynamic message
 // the default Marshaller would be used, which would strip any zero/default values from the JSON.
 // eg. given this message:
 //
 //	message Point {
 //	   double x = 1;
-//		  double y = 2;
-//		  double z = 3;
+//		double y = 2;
+//		double z = 3;
```
Comment (olegbespalov) [inline lib/netext/grpcext/stream.go:72]:
```suggestion
//		  double y = 2;
//		  double z = 3;
```

**Review by mstoykov — APPROVED** [review]

**Review by codebien — APPROVED** [review]

**Comment by olegbespalov** [comment]
For the record, I just rebased and squashed some commits, going to merge

**Pull request merged**